### PR TITLE
verify_env function for windows

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?

Creates `win_verify_env` to handle folder creation and permissions for windows
Fixes problem with missing `os.getppid` method in windows
Creates win_functions util
### What issues does this PR fix or reference?

https://github.com/saltstack/zh/issues/674
https://github.com/saltstack/zh/issues/675
### Previous Behavior

Folders were not being created on minion/master start in windows
Stacktrace when launching minion/master when minion/master is running
### New Behavior

Creates missing folders and adds permissions
Gracefully exits when instance is already running
### Tests written?

No

### CODE REVIEW

**Comment by cachedout** [comment]
Hi @twangboy Just some lint here. The rest looks great.

http://166.78.178.63:8080/job/PR/job/salt-pr-lint-n/1576/violations/

File: salt/utils/parsers.py [hunk 1]
```diff
@@ -911,9 +911,17 @@ def check_running(self):
         Check if a pid file exists and if it is associated with
         a running process.
         '''
+        # There is no os.getppid method for windows
+        from salt.utils import is_windows
+        if is_windows():
+            from salt.utils.win_functions import get_parent_pid
```
Comment (skizunov) [inline salt/utils/parsers.py]: This file already has 'import salt.utils' at the top. So I would skip the import and just use:

```
if salt.utils.is_windows():
```

File: salt/utils/win_functions.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+def __virtual__():
+    '''
+    Load only on Windows with necessary modules
+    '''
+    if not salt.utils.is_windows():
+        return False, 'This utility only works on Windows'
+    if not HAS_WIN32:
+        return False, 'This utility requires pywin32'
+
+    return 'win_runas'
```
Comment (skizunov) [inline salt/utils/win_functions.py]: I am curious, is there a reason why this is 'win_runas' and not 'win_functions'?

Comment (twangboy) [inline salt/utils/win_functions.py]: @skizunov Because I'm an idiot

File: salt/utils/parsers.py (hunk 1 above)
Comment (twangboy) [inline salt/utils/parsers.py]: That's what I thought, but it wasn't resolving for me when I was doing it earlier. It is now though. I don't know what happened.

**Comment by damon-atkins** [comment]
FYI
```
os.getppid()
Return the parent’s process id. When the parent process has exited, on Unix the id returned is the one of the init process (1), on Windows it is still the same id, which may be already reused by another process.
Availability: Unix, Windows.
Changed in version 3.2: Added support for Windows.
```

**Pull request merged**